### PR TITLE
Check usb/jtag for incoming chars too

### PR DESCRIPTION
**Pull request by tannewt**
@bill88t Please try this for #7716. I don't get any hangs with it.

### CODE REVIEW

**Comment by RetiredWizard** [comment]
I'm still seeing issues with the REPL using these artifacts. Some of my issues could be Thonny related though, Thonny definitely is confused by the QTPY C3 at times and ends up making things worse.

I think the easiest way to see the issues I'm seeing is by trying to transfer a file greater than about 5K to the microcontroller using Thonny. I don't have the same problem sending files using the web workflow interface.

Before web workflow was available, I had written my own file transfer routine which failed after about the same amount of data unless I sent one character at time and waited for the microcontroller to acknowledge receiving each character.

I'm not sure if this is the same issue that bill88t was seeing though, if his issues are resolved, I can open a new issue for this.

**Comment by bill88t** [comment]
Using tio and old ljinux 0.3.6 (latest master doesn't support incomplete `console` objects), I am unable to reproduce any of the original issues.
It can scroll in `nano`, redraw the whole screen, read and respond all the ansi escape sequences just perfectly.

Spamming spaces during boot (wifi & web workflow init) seems to have left a bit of unreadable garbage, but that's about it.
After it boots, it's rock solid.

I spent 15 minutes mashing the keyboard and using terminal applications.

**Comment by RetiredWizard** [comment]
I'll wait until this has been merged in and then re-test the issue I'm seeing. If I still see it and I can pull together a reproducible demonstration I'll open up a new issue. Really glad to see the C3 making progress!!!

File: ports/espressif/supervisor/usb_serial_jtag.c
Comment (bill88t) [inline ports/espressif/supervisor/usb_serial_jtag.c:107]: Perhaps move this up and use it for the return?
It should not make much difference since optimisations are made, but yea.

File: ports/espressif/supervisor/usb_serial_jtag.c
Comment (bill88t) [inline ports/espressif/supervisor/usb_serial_jtag.c:116]: Haven't seen extras so it should be good.

File: ports/espressif/supervisor/usb_serial_jtag.c
Comment (bill88t) [inline ports/espressif/supervisor/usb_serial_jtag.c:113]: Won't this affect gdb? I should probably test it.

Comment (tannewt) [reply ports/espressif/supervisor/usb_serial_jtag.c:113]: I don't think so since it just disables the serial out interrupt.

File: ports/espressif/supervisor/usb_serial_jtag.c
Comment (tannewt) [reply ports/espressif/supervisor/usb_serial_jtag.c:107]: Ya, it could be. I'm not sure what the advantage is though.

**Review by dhalbert — APPROVED** [review]
Looks good, but I did not test.

**Pull request merged**
